### DISCUSSION TITLE
UncheckedExecutionException: java.lang.IllegalStateException: Unable to load cache item

### DISCUSSION THREAD

**Discussion by myelkovan** [playframework/playframework#11394] [General]
Hi
I try to decide if I should learn play framework or not. I started with Hello World project but It did not work on windows 10. It said me that Java 18 is too new for Play and I downloaded Java 17 although others are waiting to upgrade Java 19. Unfortunately Hello World did not work with Java 17 and I tried to create new project following the tutorial on playframework.com. I tried Create a Play Java or Scala project using sbt new. And It did not work. I never see a framework that hello world project would cause problem for the ones who are new and ready to find something else. This is the problem when I go on http://localhost:9000/

```
[info] Version 2.8.16 running Java 17.0.4.1
[info]
[info] Play is run entirely by the community. Please consider contributing and/or donating:
[info] https://www.playframework.com/sponsors
[info]
[info] Running Play on Java 17 is experimental. Tweaks are necessary:
[info] https://github.com/playframework/playframework/releases/2.8.15
[info]

--- (Running the application, auto-reloading is enabled) ---

[info] p.c.s.AkkaHttpServer - Listening for HTTP on /[0:0:0:0:0:0:0:0]:9000

(Server started, use Enter to stop and go back to the console...)

2022-08-19 11:37:10 ERROR play.api.http.DevHttpErrorHandler

! @7oke1cg75 - Internal server error, for (GET) [/] ->

play.api.UnexpectedException: Unexpected exception[UncheckedExecutionException: java.lang.IllegalStateException: Unable to load cache item]
        at play.core.server.DevServerStart$$anon$1.reload(DevServerStart.scala:254)
        at play.core.server.DevServerStart$$anon$1.get(DevServerStart.scala:148)
        at play.core.server.AkkaHttpServer.handleRequest(AkkaHttpServer.scala:308)
        at play.core.server.AkkaHttpServer.$anonfun$createServerBinding$1(AkkaHttpServer.scala:225)
```

**Comment by Squeng** [comment]
While I don't know what the root cause(s) might be in your case, I can confirm that Play works fine with Java 17 in my case:
https://github.com/Squoss/Squawg/blob/main/beapi/build.sbt#L24

**Comment by mkurz** [comment]
In case you tried Java 18 and afterwards Java 17, did you remove the `target` folder within your project folder? There could have been class files left compiled with Java 18, so it's better to start with a clean environment.

**Comment by mamt4real** [comment]
Exactly my problem I dont even know where to begin !!!

**Comment by mreiche** [comment]
I fixed it by force using Java 11.
I installed Java 11 using brew (on MacOS)
```shell
brew install openjdk@11
```
And run setting the `JAVA_HOME` explicitly (environment variable is not supported for some reason).

```shell
sbt run --java-home "/opt/homebrew/Cellar/openjdk@11/11.0.16.1_1"
```

**Comment by mamt4real** [comment]
Thanks for the reply....and that's the exact solution that works for me I
had to uninstall my jdk17 and replace with 11

**Comment by slisaasquatch** [comment]
I just want to say that I ran into the same error with a simple Play 2.8.18 project on Java 17 with a fresh build. The error does not happen on Java 8 or 11.
